packages/project-disaster-trail/src/components/Game/TaskScreen/index.js

```diff
@@ -1,40 +1,240 @@
+    timer.reset();
+    timer.addCallback((t, p) => {
+      setPercentComplete(p);
+    });
+    timer.addCompleteCallback(() => onTimerComplete());
+    timer.start();
+  };
+
+  // when the component mounts, start a timer of the active task's time
   useEffect(() => {
-    const taskChapterTimer = setTimeout(
-      endChapter,
-      60 * 1000 /* end chapter after 1 minute */
-    );
+    if (timer && activeTask && action === ACTIONS.SOLVING) {
+      startTimer(activeTask.time);
+    }
 
     return () => {
-      clearTimeout(taskChapterTimer);
+      timer.stop();
+    };
+  }, [timer, activeTask, action]);
+
+  useEffect(() => {
+    chapterTimer.setDuration(chapterDuration);
+    chapterTimer.addCompleteCallback(() => endChapter());
+    chapterTimer.start();
+    return () => {
+      chapterTimer.stop();
     };
-  }, []);
+  }, [chapterTimer]);
+
+  useEffect(() => {
+    switch (action) {
+      case ACTIONS.SOLVING:
+        if (prevActiveTask && !activeTask) {
+          setVotingComplete(false);
+          setAction(ACTIONS.VOTING);
+        }
+        break;
+      case ACTIONS.VOTING:
+        if (votingComplete) {
+          setMovingMapComplete(false);
+          setAction(ACTIONS.MOVING_MAP);
+        }
+        break;
+      case ACTIONS.MOVING_MAP:
+        if (movingMapComplete) {
+          setAction(ACTIONS.SOLVING);
+        }
+        break;
+      default:
+        // eslint-disable-next-line no-console
+        console.warn("Unknown action: ", action);
+        break;
+    }
+  }, [prevActiveTask, activeTask, votingComplete, movingMapComplete]);
+
+  // useEffect(() => {
+  //   if (activeTask && activeTask !== prevActiveTask) {
+  //     // eslint-disable-next-line no-console
+  //     console.log("new task");
+  //   }
+  // }, [activeTask, prevActiveTask]);
+
+  // when the user transitions from one action to another
+  useEffect(() => {
+    if (action !== prevAction) {
+      switch (action) {
+        case ACTIONS.SOLVING:
+          break;
+        case ACTIONS.VOTING:
+          startTimer(votingDuration);
+          break;
+        case ACTIONS.MOVING_MAP:
+          startTimer(mapTransitionDuration);
+          break;
+        default:
+          // eslint-disable-next-line no-console
+          console.log("unknown action ", action);
+      }
+    }
+  }, [action, prevAction]);
+
+  const onItemSelection = item => {
+    // eslint-disable-next-line no-console
+    console.log("item ", item);
+    // if (item.type === activeTask.requiredItem) {
+    //   this.setState(state => ({
+    //     correctItemsChosen: state.correctItemsChosen + 1
+    //   }));
+    // }
+  };
+
+  const onTaskSelection = task => {
+    // eslint-disable-next-line no-console
+    console.log("onTaskSelection ", task);
+    // const { taskVotes } = this.state;
+    // taskVotes[task.type] += 1;
+
+    // this.setState({
+    //   taskVotes
+    // });
+    // // Return true so Orb knows how to animate
+    // return true;
+  };
```

Comment:
If the orb manager was showing tasks to choose between, voting still wouldn't work

Reply:
This also shouldn't be hard to solve

Reply:
Yup, we'll fix that too post-merge